File: themes/lib/targets/ghostty.py
"""Ghostty terminal color theme generator."""

from themes.lib.schema import ColorScheme, ThemeState

TARGET_NAME = "ghostty"
ASSEMBLY = "concat"
OUTPUT_PATH = "~/.config/ghostty/config"
BASE_PATH = "~/repos/dotfiles/config/ghostty/base"
RELOAD_CMD = None  # Ghostty watches config files and auto-reloads
COMMENT = "#"


def generate(colors: ColorScheme, state: ThemeState) -> str:
    lines = [
        f"font-family = {state.mono_font}",
        f"font-size = {state.mono_font_size_for(TARGET_NAME)}",
        f"background = {colors.bg}",
        f"foreground = {colors.fg}",
        f"selection-background = {colors.bg3}",
        f"selection-foreground = {colors.fg}",
        f"cursor-color = {colors.fg}",
        f"cursor-text = {colors.bg}",
    ]
    for i, color in enumerate(colors.palette):
        lines.append(f"palette = {i}={color}")
    return "\n".join(lines) + "\n"

File: desktopctl/src/theme/targets/ghostty.rs
use super::{Assembly, GeneratedContent, TargetMetadata};
use crate::theme::schema::{ColorScheme, ThemeState};

pub const METADATA: TargetMetadata = TargetMetadata {
    name: "ghostty",
    assembly: Assembly::Import,
    output_path: Some("~/.config/ghostty/theme.conf"),
    base_path: None,
    extra_outputs: &[],
    managed_paths: &[],
    reload_cmd: None,
    comment: Some("#"),
    sync_safe: true,
};

pub fn generate(colors: &ColorScheme, state: &ThemeState) -> crate::Result<GeneratedContent> {
    let mut lines = vec![
        format!("font-family = {}", state.mono_font),
        format!("font-size = {}", state.mono_font_size_for(METADATA.name)?),
        format!("background = {}", colors.bg),
        format!("foreground = {}", colors.fg),
        format!("selection-background = {}", colors.bg3),
        format!("selection-foreground = {}", colors.fg),
        format!("cursor-color = {}", colors.fg),
        format!("cursor-text = {}", colors.bg),
    ];
    for (index, color) in colors.palette.iter().enumerate() {
        lines.push(format!("palette = {index}={color}"));
    }
    Ok(GeneratedContent::text(format!("{}\n", lines.join("\n"))))
}
